### PR TITLE
1.1.0

### PR DESCRIPTION
**Pull request by andrasguseo**

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -64,30 +64,38 @@ public function init() {
...
+		 * @return array
 		 */
-		public function filter_ical_query( $query ) {
+		public function filter_ical_query( $args ) {
 
-			if ( ! isset( $_GET['ical'] )
-			     || ! isset( $query->tribe_is_event_query )
-			     || ! $query->tribe_is_event_query ) {
+			// If ical is not set in the URL then bail
+			if ( ! isset( $_GET['ical'] ) || $_GET['ical'] != 1 ) {
```
Comment (lucatume) [inline index.php]: This too should be sanitized before checking.
Not to waste a check, though, user `filter_input_array`:

```php
$filters = [
    'ical' => FILTER_SANITIZE_INT,
    'tribe_display' => FILTER_SANITIZE_STRING,
    'start_date' => FILTER_SANITIZE_STRING,
    'end_date' => FILTER_SANITIZE_STRING,
];
$vars = filter_input_array( INPUT_GET, $filters );
```

The code can then use the safe `$vars` variable:

```php
if( ! isset( $vars ['ical'] ) { //... }
```

File: index.php [hunk 2]
```diff
@@ -64,30 +64,38 @@ public function init() {
...
 				return;
 			}
 
-			$tribe_display = $_GET[ 'tribe_display' ];
-			$start_date    = isset( $_GET[ 'start_date' ] ) ? $_GET[ 'start_date' ] : '';
-			$end_date      = isset( $_GET[ 'end_date' ] )   ? $_GET[ 'end_date' ]   : '';
+			$tribe_display = isset( $_GET['tribe_display'] )
```
Comment (lucatume) [inline index.php]: All of this sanitization logic could be moved up.

File: index.php [hunk 3]
```diff
@@ -64,30 +64,38 @@ public function init() {
...
+				: '';
+			$end_date      = isset( $_GET['end_date'] )
+				? filter_var( $_GET['end_date'], FILTER_SANITIZE_STRING )
+				: '';
 
 			if ( $tribe_display === 'custom' ) {
 
 				// Check if there is a start_date set
-				if ( isset( $start_date ) && !empty( $start_date ) ) {
+				if ( isset( $start_date ) && ! empty( $start_date ) ) {
```
Comment (lucatume) [inline index.php]: If the sanitization logic is moved up, then these would be checks on `$vars`.

**Review by lucatume — CHANGES_REQUESTED** [review]
I think the code could be improved by the use of the `filter_input_array` function; it was made exactly for this type of case.

File: index.php (hunk 1 above)
Comment (andrasguseo) [reply index.php]: For the number it was 
`FILTER_SANITIZE_NUMBER_INT`
but all good. This is nice.

**Review by lucatume — APPROVED** [review]

**Comment by lucatume** [comment]
I would like to say I knew... Yet I just forgot, sorry.

**Pull request merged**